Tensile/Source/lib/include/Tensile/PropertyMatching.hpp

```diff
@@ -624,5 +624,276 @@ namespace Tensile
                            : std::make_tuple(this->nullValue, std::numeric_limits<double>::max());
             }
         };
+
+        /**
+         * Specialization of DistanceMatchingTable for GridBased Distance. This special case will
+         * use a more efficient search algorithm that depends on the grid structure
+         */
+        template <typename Key, typename Object, typename Value, typename ReturnValue>
+        struct DistanceMatchingTable<Key,
+                                     Object,
+                                     Value,
+                                     ReturnValue,
+                                     Matching::GridBasedDistance<Key>>
+            : public DistanceMatchingCommon<Key,
+                                            Object,
+                                            Value,
+                                            ReturnValue,
+                                            Matching::GridBasedDistance<Key>>
+        {
+            using Base              = MatchingTable<Object, Value, ReturnValue>;
+            using Entry             = MatchingTableEntry<Key, Value>;
+            using Transform         = typename Base::Transform;
+            using Properties        = typename Base::Properties;
+            using GridBasedDistance = Matching::GridBasedDistance<Key>;
+            using Common            = DistanceMatchingCommon<Key,
+                                                  Object,
+                                                  Value,
+                                                  ReturnValue,
+                                                  Matching::GridBasedDistance<Key>>;
+            using Common::distance;
+            using Common::nullValue;
+            using Common::table;
+
+            DistanceMatchingTable(ReturnValue nullValue = ReturnValue())
+                : Common(nullValue)
+            {
+            }
+
+            DistanceMatchingTable(Properties const& properties,
+                                  ReturnValue       nullValue = ReturnValue())
+                : Common(properties, nullValue)
+            {
+            }
+
+            DistanceMatchingTable(GridBasedDistance const& distance,
+                                  Properties const&        properties,
+                                  ReturnValue              nullValue = ReturnValue())
+                : Common(distance, properties, nullValue)
+            {
+            }
+
+            std::tuple<ReturnValue, double> findBestKeyMatch(Key const& key,
+                                                             Transform  transform) const
+            {
+                if(Debug::Instance().printPropertyEvaluation())
+                    return findBestKeyMatch_GridBased<true>(key, transform);
+                else
+                    return findBestKeyMatch_GridBased<false>(key, transform);
+            }
+
+            template <bool T_Debug>
```

Comment:
Howcome you have the debug param passed in as a template? It might make the function sig a bit cleaner to just do 

`bool T_Debug = Debug::Instance().printPropertyEvaluation()`

at the start of the function rather

Reply:
This design is used for the other search algorithms, so it's mostly just used here for consistency. I'm guessing it was done this way to make the non-debug path have less branching/checking the debug flag, but I have no idea if it actually has a measurable impact on runtime.